### DISCUSSION TITLE
Can't access column from aggregates

### DISCUSSION THREAD

**Discussion by virendrakabra14** [andialbrecht/sqlparse#759] [General]
For this SQL
```sql
SELECT COUNT(tab1.col1) AS cnt FROM tab1
```

How do I access `tab1.col1` that is inside the count-aggregate?

`node = parsed[0].tokens[2]` gives the `Identifier` node but I cannot proceed after that.
`node.get_real_name()` gives `COUNT` and `node.get_alias()` gives `cnt`

**Comment by andialbrecht** [comment]
Hi @virendrakabra14,

when working on that low level the `_pprint_tree()` (hidden) helper function comes in handy. Given your example:

```
>>> import sqlparse
>>> p = sqlparse.parse("SELECT COUNT(tab1.col1) AS cnt FROM tab1")[0]
>>> p._pprint_tree()
|- 0 DML 'SELECT'
|- 1 Whitespace ' '
|- 2 Identifier 'COUNT(...'
|  |- 0 Function 'COUNT(...'
|  |  |- 0 Identifier 'COUNT'
|  |  |  `- 0 Name 'COUNT'
|  |  `- 1 Parenthesis '(tab1....'
|  |     |- 0 Punctuation '('
|  |     |- 1 Identifier 'tab1.c...'
|  |     |  |- 0 Name 'tab1'
|  |     |  |- 1 Punctuation '.'
|  |     |  `- 2 Name 'col1'
|  |     `- 2 Punctuation ')'
|  |- 1 Whitespace ' '
|  |- 2 Keyword 'AS'
|  |- 3 Whitespace ' '
|  `- 4 Identifier 'cnt'
|     `- 0 Name 'cnt'
|- 3 Whitespace ' '
|- 4 Keyword 'FROM'
|- 5 Whitespace ' '
`- 6 Identifier 'tab1'
   `- 0 Name 'tab1'
>>> p.tokens[2].tokens[0].tokens[1].tokens[1].get_real_name()
'col1'
>>>
```

`sqlparse` hasn't very useful functions for inspecting a SQL statement in a more human way yet. That's why you have to get down the whole way to the identifier, in fact you have to know the concrete statement - which isn't really satisfying ;-)
